src/lang/es.rs

```diff
@@ -0,0 +1,1044 @@
+            let tens = ((triplet / 10) % 10) as usize;
+            let units = (triplet % 10) as usize;
+
+            if hundreds > 0 {
+                match triplet {
+                    // Edge case when triplet is a hundred
+                    100 => words.push(String::from("cien")),
+                    _ => words.push(String::from(CENTENAS[hundreds])),
+                }
+            }
+
+            if tens != 0 || units != 0 {
+                let unit_word = match (units, i) {
+                    // case `1_100` => `mil cien` instead of `un mil cien`
+                    // case `1_001_000` => `un millón mil` instead of `un millón un mil`
+                    // Explanation: Second triplet is always read as thousand, so we
+                    // don't need to say "un mil"
+                    (_, 1) if triplet == 1 => "",
+                    // case `001_001_100...` => `un billón un millón cien mil...` instead of
+                    // `uno billón uno millón cien mil...`
+                    // All `triplets == 1`` can can be named as "un". except for first or second
+                    // triplet
+                    (_, index) if index != 0 && triplet == 1 => "un",
+                    _ => UNIDADES[units],
+                };
+
+                match tens {
+                    // case `?_102` => `? ciento dos`
+                    0 => words.push(String::from(unit_word)),
+                    // case `?_119` => `? ciento diecinueve`
+                    // case `?_110` => `? ciento diez`
+                    1 => words.push(String::from(DIECIS[units])),
+                    2 if self.prefer_veinte && units != 0 => {
+                        let unit_word = if units == 1 && i != 0 { "un" } else { unit_word };
+                        words.push(format!("veinte y {unit_word}"));
+                    }
+                    2 => words.push(match units {
+                        0 => String::from(DECENAS[tens]),
+                        // case `021_...` => `? veintiún...`
+                        1 if i != 0 => String::from("veintiún"),
+                        // case `?_021` => `? veintiuno`
+                        _ => format!("veinti{unit_word}"),
+                    }),
+                    _ => {
+                        // case `?_142 => `? ciento cuarenta y dos`
+                        let ten = DECENAS[tens];
+                        words.push(match units {
+                            0 => String::from(ten),
+                            _ => format!("{ten} y {unit_word}"),
+                        });
+                    }
+                }
+            }
+
+            // Add the next Milliard if there's any.
+            if i != 0 && triplet != 0 {
+                if i > MILLARES.len() - 1 {
+                    return Err(Num2Err::CannotConvert);
+                }
+                // Boolean that checks if next Milliard is plural
+                let plural = triplet != 1;
+                match plural {
```

Comment:
you should probably use a `if` case here rather than a `match` for bool (`if plural { ... } else { ... }`)

Reply:
I figured it looked slimmer with match instead of if
```rust
// preview:
if plural {
    words.push(String::from(MILLARES[i]))
} else {
    words.push(String::from(MILLAR[i]))
}

// vs match
match plural {
    false => words.push(String::from(MILLAR[i])),
    true => words.push(String::from(MILLARES[i])),
}
```